src/server/pfs/server/driver.go

```diff
@@ -2194,12 +2149,124 @@ func (d *driver) getTreeForCommit(ctx context.Context, commit *pfs.Commit) (hash
 	return h, nil
 }
 
+func (d *driver) getTree(ctx context.Context, commitInfo *pfs.CommitInfo, path string) (rs []*hashtree.Reader, cleanup func() error, retErr error) {
+	// Setup cleanup function
+	var f *os.File
+	cleanup = func() error {
+		return os.Remove(f.Name())
+	}
+	pachClient := d.getPachClient(ctx)
+	objClient, err := obj.NewClientFromEnv(ctx, d.storageRoot)
+	if err != nil {
+		return nil, cleanup, err
+	}
+	// Determine the hashtree in which the path is located and download the chunk it is in
+	idx := hashtree.PathToTree(path, int64(len(commitInfo.Trees)))
+	if f, err = d.downloadTree(ctx, pachClient, objClient, commitInfo.Trees[idx], path); err != nil {
+		return nil, cleanup, err
+	}
+	return []*hashtree.Reader{hashtree.NewReader(f, nil)}, cleanup, nil
+}
+
+func (d *driver) getTrees(ctx context.Context, commitInfo *pfs.CommitInfo, pattern string) (rs []*hashtree.Reader, cleanup func() error, retErr error) {
```

Comment:
The signature of this function seems a little bit weird because it's returning a `cleanup` function that needs to be called to cleanup the state of the `hashtree.Reader`s I would expect the type to be `hashtree.ReadCloser`s and satisfy the `ReadCloser` interface. That seems a little more idiomatic.